pkg/scheduler/framework/plugins/interpodaffinity/plugin.go

```diff
@@ -132,3 +134,102 @@ func GetNamespaceLabelsSnapshot(logger klog.Logger, ns string, nsLister listersv
+		}
+		if len(antiTerms) != 0 && podMatchesAllAffinityTerms(antiTerms, originalPod) && !podMatchesAllAffinityTerms(antiTerms, modifiedPod) {
+			updateFromMatchPodAntiAffinityToNotMatch = true
+		}
+		if (len(terms) != 0 && !updateFromNotMatchPodAffinityToMatch) || (len(antiTerms) != 0 && !updateFromMatchPodAntiAffinityToNotMatch) {
+			logger.V(5).Info("a scheduled pod was updated but it doesn't match the target pod's affinity or does match the target pod's anti-affinity",
+				"pod", klog.KObj(pod), "modifiedPod", klog.KObj(modifiedPod))
+			return framework.QueueSkip, nil
+		}
+		logger.V(5).Info("a scheduled pod was updated and it matches the target pod's affinity or doesn't match the target pod's anti-affinity, and the pod may be schedulable now",
+			"pod", klog.KObj(pod), "modifiedPod", klog.KObj(modifiedPod))
+		return framework.Queue, nil
+	}
+
+	// Pod is added. Return Queue when the added pod matching the target pod's affinity.
+	if modifiedPod != nil {
+		if podMatchesAllAffinityTerms(terms, modifiedPod) {
+			logger.V(5).Info("a scheduled pod was added and it matches the target pod's affinity",
+				"pod", klog.KObj(pod), "modifiedPod", klog.KObj(modifiedPod))
+			return framework.Queue, nil
+		}
+		logger.V(5).Info("a scheduled pod was added and it doesn't match the target pod's affinity",
+			"pod", klog.KObj(pod), "modifiedPod", klog.KObj(modifiedPod))
+		return framework.QueueSkip, nil
+	}
+
+	// Pod is deleted. Return Queue when the deleted pod matching the target pod's anti-affinity.
+	for _, t := range antiTerms {
+		if t.Matches(originalPod, nil) {
+			logger.V(5).Info("a scheduled pod was deteled but it matches the target pod's anti-affinity",
+				"pod", klog.KObj(pod), "modifiedPod", klog.KObj(modifiedPod))
+			return framework.QueueSkip, nil
+		}
+	}
+	logger.V(5).Info("a scheduled pod was deleted but it doesn't match the target pod's anti-affinity, and the pod may be schedulable now",
+		"pod", klog.KObj(pod), "modifiedPod", klog.KObj(modifiedPod))
+	return framework.Queue, nil
+}
+
+func (pl *InterPodAffinity) isSchedulableAfterNodeChange(logger klog.Logger, pod *v1.Pod, oldObj, newObj interface{}) (framework.QueueingHint, error) {
+	originalNode, modifiedNode, err := util.As[*v1.Node](oldObj, newObj)
+	if err != nil {
+		return framework.Queue, err
+	}
+
+	terms, err := framework.GetAffinityTerms(pod, framework.GetPodAffinityTerms(pod.Spec.Affinity))
+	if err != nil {
+		return framework.Queue, err
+	}
+
+	// Return Queue after a new node was added.
+	if originalNode == nil {
+		for _, term := range terms {
+			if _, ok := modifiedNode.Labels[term.TopologyKey]; !ok {
+				logger.V(5).Info("a new node without matched topologyKey was added", "pod", klog.KObj(pod), "node", klog.KObj(modifiedNode))
+				return framework.QueueSkip, nil
+			}
+		}
+		logger.V(5).Info("the pod may be schedulable now because a new node was added", "pod", klog.KObj(pod), "node", klog.KObj(modifiedNode))
+		return framework.Queue, err
```

Comment:
If a Node has one or more than one label which is the same as TopologyKey, it should return Queue, right?

```suggestion
		for _, term := range terms {
			if _, ok := modifiedNode.Labels[term.TopologyKey]; ok {
				logger.V(5).Info("a new node with topologyKey was added and it may make pod schedulable", "pod", klog.KObj(pod), "node", klog.KObj(modifiedNode))
				return framework.Queue, nil
			}
		}
		logger.V(5).Info("a new node is added but doesn't have any topologyKey of pod affinity", "pod", klog.KObj(pod), "node", klog.KObj(modifiedNode))
		return framework.QueueSkip, err
```